src/test/test_ptr_slow.c

```diff
@@ -0,0 +1,103 @@
+/* Copyright (c) 2001-2004, Roger Dingledine.
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2019, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+#include "orconfig.h"
+#include "core/or/or.h"
+#include "test/test.h"
+#include "test/ptr_helpers.h"
+
+#include <stdint.h>
+#include <limits.h>
+
+/** Assert that <b>a</b> can be cast to void * and back. */
+static void
+assert_int_voidptr_roundtrip(int a)
```

Comment:
Is sizeof(int) always smaller than or equal to sizeof(void *) on all our supported platforms?
I think they are. But I'm not sure if that's guaranteed by the C spec.

Do we need to document this assumption, maybe using the SIZEOF macros?
Or does Tor already do a similar size check elsewhere?

Reply:
We can leverage the build system for this. See: 13b28063f9f2a3d2a587eabac358452a42277052